lib/logstash/inputs/rabbitmq.rb

```diff
@@ -131,17 +139,87 @@ def consume!
         end
       end
 
+      public
       def stop
         super
         shutdown_consumer
         close_connection
       end
 
+      public
       def shutdown_consumer
         return unless @consumer
         @consumer.gracefully_shut_down
       end
 
+      private
+      def get_header_value(value)
+        # Two kinds of values require exceptional treatment:
+        #
+        # String values are instances of
+        # com.rabbitmq.client.impl.LongStringHelper.ByteArrayLongString
+        # and we don't want to propagate those.
+        #
+        # List values are java.util.ArrayList objects and we need to
+        # recurse into them to convert any nested strings values.
+        if value.respond_to? "toArray"
```

Comment:
I don't know what the performance costs are, but my instinct is that a class check is quicker than reflection. I also feel that a class check is more explicit about the intent of behavior and expectations, but others can feel differently ;)